client/keys/show.go

```diff
@@ -30,16 +39,20 @@ func getKey(name string) (keys.Info, error) {
 // CMD
 
 func runShowCmd(cmd *cobra.Command, args []string) error {
-	if len(args) != 1 || len(args[0]) == 0 {
-		return errors.New("You must provide a name for the key")
+	info, err := getKey(args[0])
+	if err != nil {
+		return err
 	}
-	name := args[0]
-
-	info, err := getKey(name)
-	if err == nil {
-		printInfo(info)
+	if viper.GetBool(flagExportPubKey) {
+		out, err := info.PubKey.MarshalJSON()
+		if err != nil {
+			return err
+		}
+		fmt.Println(string(out))
```

Comment:
I don't like the idea of moving out of the current process train of printing everything through the `printInfo` function (right below this comment)...
Also don't like introducing another key. Already you can get the pubkey in json by using `gaiacli keys show <yourkey> --output json` - If you want to also see the pubkey with "text" output (aka `--output text` aka the default) then we should add that right about here: https://github.com/cosmos/cosmos-sdk/blob/18ac0f99d152c47bf70b8c749ee2e0b49d9dcbde/client/keys/utils.go#L58

maybe with a table header so it would look like:
```
name          address                pubkey 
alessio        197609375...       0x1241412.... 
```
 - just need to make sure we'd get it right for `keys list` as well as `keys show` but meh - I'm okay with just using `--output json` myself

Reply:
Hi!

> Already you can get the pubkey in json by using gaiacli keys show <yourkey> --output json
True, but the following prints the private key out too:

```
alessio@bizet:~/.../github.com/tendermint/clearchain$ clearchainctl keys show --output json ch_admin_1
{"name":"ch_admin_1","pubkey":[1,"CEB016A0495DEE269D655F12DE14941963214B3BEE034C3DDFDBCECB2627D3C7"],"privkey.armor":"-----BEGIN TENDERMINT PRIVATE KEY-----\nkdf: bcrypt\nsalt: CB69DDE1B83E179E7DED2C9A7FDE1BA8\n\n3Q2YAte6PArOEnMt9YR5CNmAyk7H6cFvZLuibwxoIlf+B7whJ8Xj4TZmBDo/JfEE\nlcdUZTafvGvHp2vxZ7PVg0ria+sJIPqQlgaNI1iGXqZ/kRYHSuRPIKwniRwrk6LE\nXFE5c8DN3Ceu\n=v8dc\n-----END TENDERMINT PRIVATE KEY-----"}
```

Furthermore, the public key JSON format seems a bit fuzzy: `[1,"CEB016A0495DEE269D655F12DE14941963214B3BEE034C3DDFDBCECB2627D3C7"]`

`If you want to also see the pubkey with "text" output (aka --output text aka the default) then we should add that right about here:
[snip]
`

That's fair, I'd be happy with that too! I just need to allow users to exchange public keys in a portable format, be it a string or a JSON thing.

Reply:
So it sounds like we just want to improve the JSON output format, which I could totally get behind - maybe don't print the private key by default too

Reply:
Okay yeah the validator json key is not a HEX string which stinks